File: server-go/internal/discovery/beacon.go
package discovery

import "encoding/json"

const (
	BeaconPort       = 8000
	BeaconAddr       = "255.255.255.255"
	BeaconIntervalMs = 2000
)

func BeaconMessage(name string, port int) string {
	b, _ := json.Marshal(map[string]any{"app": "cardputerme", "v": 1, "name": name, "port": port})
	return string(b)
}

File: server/lib/beacon.js
'use strict';

const BEACON_PORT = 8000;
const BEACON_ADDR = '255.255.255.255';
const BEACON_INTERVAL_MS = 2000;

function beaconMessage(session, port) {
  return JSON.stringify({ app: 'cardputerme', session, port });
}

module.exports = { beaconMessage, BEACON_PORT, BEACON_ADDR, BEACON_INTERVAL_MS };
